### PR TITLE
Logan/merge next

### PR DESCRIPTION
**Pull request by logan-markewich**

### CODE REVIEW

File: llama-index-legacy/tests/ingestion/test_pipeline.py
[the remark this answers is not in the document]
Comment (logan-markewich) [reply llama-index-legacy/tests/ingestion/test_pipeline.py:1]: I didn't want any diff with legacy

[the remark this answers is not in the document]
Comment (logan-markewich) [reply llama-index-legacy/tests/ingestion/test_pipeline.py:1]: I can add back to core

File: llama-index-integrations/indices/llama-index-indices-managed-llama-cloud/llama_index/indices/managed/llama_cloud/base.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        self,
+        name: str,
+        nodes: Optional[List[BaseNode]] = None,
+        transformations: Optional[List[TransformComponent]] = None,
+        timeout: int = 60,
+        project_name: str = DEFAULT_PROJECT_NAME,
+        api_key: Optional[str] = None,
+        base_url: Optional[str] = None,
+        app_url: Optional[str] = None,
+        service_context: Optional[ServiceContext] = None,
```
Comment (Disiok) [inline llama-index-integrations/indices/llama-index-indices-managed-llama-cloud/llama_index/indices/managed/llama_cloud/base.py:40]: thoughts about removing service context support here?

File: llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py
Comment (Disiok) [inline llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py:1]: older description seem more detailed here.

File: llama-index-integrations/embeddings/llama-index-embeddings-azure-openai/llama_index/embeddings/azure_openai/base.py [hunk 2]
```diff
@@ -22,6 +22,11 @@ class AzureOpenAIEmbedding(OpenAIEmbedding):
         default=None, description="The Azure deployment to use."
     )
 
+    api_base: str = Field(default="", description="The base URL for Azure deployment.")
+    api_version: str = Field(
+        default="", description="The version for Azure OpenAI API."
+    )
+
```
Comment (Disiok) [inline llama-index-integrations/embeddings/llama-index-embeddings-azure-openai/llama_index/embeddings/azure_openai/base.py:29]: is this an expected addition.

File: llama-index-core/llama_index/core/callbacks/base.py [hunk 3]
```diff
@@ -139,6 +139,10 @@ def set_handlers(self, handlers: List[BaseCallbackHandler]) -> None:
         """Set handlers as the only handlers on the callback manager."""
         self.handlers = handlers
 
+    @classmethod
+    def __modify_schema__(cls, schema: Dict[str, Any]) -> None:
+        schema.update(type="object", default={})
+
```
Comment (Disiok) [inline llama-index-core/llama_index/core/callbacks/base.py:145]: what was the motivation for this again? a comment would help.

**Review by Disiok — APPROVED** [review]

File: llama-index-core/llama_index/core/evaluation/eval_utils.py [hunk 4]
```diff
@@ -61,6 +71,100 @@ def get_results_df(
...
+                cur_dataset = dataset
+                break
+
+    # either create new dataset or use existing one
+    if cur_dataset is None:
+        eval_dataset = client.project.create_eval_dataset_for_project(
+            project_id=project.id, name=dataset_name
+        )
+    else:
+        eval_dataset = cur_dataset
```
Comment (sourabhdesai) [inline llama-index-core/llama_index/core/evaluation/eval_utils.py:145]: note to self - we should create a backend upsert endpoint for this

File: llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py
Comment (logan-markewich) [reply llama-index-integrations/embeddings/llama-index-embeddings-huggingface/llama_index/embeddings/huggingface/base.py:1]: too detailed for the frontend to display lol

File: llama-index-integrations/embeddings/llama-index-embeddings-azure-openai/llama_index/embeddings/azure_openai/base.py (hunk 2 above)
Comment (logan-markewich) [reply llama-index-integrations/embeddings/llama-index-embeddings-azure-openai/llama_index/embeddings/azure_openai/base.py:29]: yes, needed for frontend

File: llama-index-core/llama_index/core/callbacks/base.py (hunk 3 above)
Comment (logan-markewich) [reply llama-index-core/llama_index/core/callbacks/base.py:145]: avoids pydantic serialization issues I think. Will add a note

File: llama-index-legacy/tests/ingestion/test_pipeline.py
[the remark this answers is not in the document]
Comment (logan-markewich) [reply llama-index-legacy/tests/ingestion/test_pipeline.py:1]: confirmed these tests are still in core

File: .github/workflows/unit_test.yml [hunk 5]
```diff
@@ -8,6 +8,8 @@ on:
 
 env:
   POETRY_VERSION: "1.6.1"
+  LLAMA_CLOUD_BASE_URL: ${{ secrets.PLATFORM_BASE_URL }}
```
Comment (Disiok) [inline .github/workflows/unit_test.yml:11]: do we need this? assuming we are not running integration tests here, seem extra.

Comment (logan-markewich) [reply .github/workflows/unit_test.yml:11]: yea thats stay stuff. We can remove for now

**Pull request merged**